addons/pos_restaurant/static/src/app/models/restaurant_table.js

```diff
@@ -63,7 +63,7 @@ export class RestaurantTable extends Base {
         };
     }
     getOrder() {
-        return this["<-pos.order.table_id"][0];
+        return this.parent_id?.getOrder?.() || this["<-pos.order.table_id"][0];
```

Comment:
> The `get_order` function returns an arbitrary order, in this case index 0 and it do not take into account children table.
> 
> It would be better to have a function that returns all orders in the table group.
> 
> You can also change the name of the get_order function to be more explicit about its return value. for example `table_has_order`
> 
> Below is an example of a function that could be useful in the future
> 
> ```js
> get tableHasOrder() {
>     return this.getAllOrders().length > 0;
> }
> getTableGroupIds() {
>     const getAllParentIds = (table) => {
>         if (!table.parent_id) {
>             return [];
>         }
>         return [table.parent_id.id, ...getAllParentIds(table.parent_id)];
>     };
> 
>     const getAllChildIds = (table) => {
>         const children = table["<-restaurant.table.parent_id"] || [];
>         return children.flatMap((child) => [child.id, ...getAllChildIds(child)]);
>     };
> 
>     return new Set([this.id, ...getAllParentIds(this), ...getAllChildIds(this)]);
> }
> getAllOrders() {
>     const ids = this.getTableGroupIds();
>     return this.models["pos.order"].filter((o) => ids.has(o.table_id?.id));
> }
> ```

> The get_order function returns an arbitrary order, in this case index 0 
The idea is that tables should have only one order at any single time now. No more multiple orders per table. That's why taking the 0th index makes sense, because there should only be at most one order.
>  it do not take into account children table.
Child tables shouldn't have any orders on them. When you link 2 tables, the child table will have it's order merged with the parent table's order and this combined order will live on the parent table.

> You can also change the name of the get_order function to be more explicit about its return value
I think that the return type is quite explicit: `get_order` will return an order.

Reply:
If you think we don't need this kind of method, just rename get_order to explicitly said that we just need to check if there is an order.

Reply:
Han ok I wasn't aware of this change. then its good

Reply:
if i need to just check i can do `Boolean(table.get_order())`. I could indeed add a helper for this 
```js
hasOrder() {
    return Boolean(this.get_order());
}```